### ISSUE TITLE
OpenVPN Sensor

### ISSUE THREAD

**Issue by SmartM-ui** [travisghansen/hass-pfsense#43]
Hi,
would it be possible to have a sensor that indicates the status of the VPN connection, in case some client connected?

Thanks

**Comment by travisghansen** [comment]
Welcome! Something indicating the number of connected clients perhaps?

**Comment by SmartM-ui** [comment]
> 

Thanks for the quick answer!
I would like to have a sensor that indicates the connection of a client in order to create an automation with warning.

Congratulations on the job

**Comment by SmartM-ui** [comment]
> 

I just need to know when any client connects

**Comment by travisghansen** [comment]
Ok, and do you need to be notified every connect, or only if there’s at least 1 connection? ie: just when the number of connections goes from 0 to > 0, or any time the number of connections change?

**Comment by SmartM-ui** [comment]
Sorry if I betrayed in answering.
Even > 0 is enough

Sorry if I betrayed in answering.
It would be better if it were possible every time the number of connections changes, but even > 0 is fine too

**Comment by travisghansen** [comment]
Yeah, I think the most likely solution here would be to have a sensor with the number of currently connected openvpn clients. From there you can use automation rules based on state changes to do whatever you need from there.

Note that this solution would result in situation where the number of connected clients may remain the same (between poll intervals) but the actual connected clients may differ. For example you may have 10 connected clients and client a drops but client z connects to the number on the sensor doesn't actually change.

**Comment by SmartM-ui** [comment]
> Yeah, I think the most likely solution here would be to have a sensor with the number of currently connected openvpn clients. From there you can use automation rules based on state changes to do whatever you need from there.
> 
> Note that this solution would result in situation where the number of connected clients may remain the same (between poll intervals) but the actual connected clients may differ. For example you may have 10 connected clients and client a drops but client z connects to the number on the sensor doesn't actually change.

Thanks for the info,
I look forward to this new possibility!

**Comment by SmartM-ui** [comment]
Hi @travisghansen,
How is it going? Were you able to create a sensor for OpenVPN?
If you need a beta tester I am available.
THANK YOU!

**Comment by travisghansen** [comment]
A Christmas miracle! https://github.com/travisghansen/hass-pfsense/commit/0cc60fd02374d718c810cc85c97fef24c71ccaef

Note the sensors are disabled by default so you'll need to explicitly enable them. Give it a try and let me know how it goes.

**Comment by SmartM-ui** [comment]
> A Christmas miracle! [0cc60fd](https://github.com/travisghansen/hass-pfsense/commit/0cc60fd02374d718c810cc85c97fef24c71ccaef)
> 
> Note the sensors are disabled by default so you'll need to explicitly enable them. Give it a try and let me know how it goes.

You are Santa Claus!

Thanks, I try and inform :-)

**Comment by SmartM-ui** [comment]
HI @travisghansen ,
I activated the openvpn switch and I activated the sensors:
sensor.pfsense_dhcp_leases_offline
sensor.pfsense_dhcp_leases_online
sensor.pfsense_dhcp_leases_total

but when I connect to the VPN the number is always 0

Do I need to activate some other sensor to enable viewing of connected clients?

Thanks

**Comment by travisghansen** [comment]
Those are dhcp leases, not openvpn stats :) Per-server you will find 5 sensors:

- `connected_client_count`
- `total_bytes_{sent/recv}`
- `total_bytes_{sent/recv}_kilobytes_per_second`

**Comment by SmartM-ui** [comment]
> 

Hi @travisghansen ,
nice to read you!

I downloaded version 0.3.1 and I have 100 sensors available, but I can't find the ones you indicate for the vpn.

**Comment by travisghansen** [comment]
Yeah, you need to install `main` to get the OpenVPN stats currently. I'll have a new release shortly with the code after you are able to do some testing and confirm it works.

**Comment by SmartM-ui** [comment]
> 

Hi @travisghansen ,

Connected Clients = OK --- GREAT
I noticed that if 2 clients with the same account are connected (for example my smartphone + my notebook) there is only one client connected and not 2.

Tomorrow I will perform further tests from outside the home.

Good night

**Comment by travisghansen** [comment]
Those should be 2 separate connections/clients. In the pfSense OpenVPN status page do you see an entry per client?

If so you may need to just wait for the scan interval to go so it will pick up the correct count.

**Comment by SmartM-ui** [comment]
> 

I waited more than 5 minutes with two clients connected and I updated the home assistant pagian, just to be safe.

Tomorrow I feel better, being away from home.
Just for information, I use the same account on both notebook and smartphone, I don't know if this could be a problem.

If it were possible, it would be better to have the count of connected clients and not of the account so if someone were to try to log in with my same account, I would still know that there is another device connected.

I don't know if I've made myself clear :-)

THANK YOU again, goodnight

**Comment by travisghansen** [comment]
Yeah, when connected with both your devices go into the pfSense OpenVPN status page and let me know if you see 1 or 2 entries there.

**Comment by SmartM-ui** [comment]
> 

Hi @travisghansen ,
I found the problem and it wasn't your integration.

By default, pfSense blocks concurrent connections from the same user.
By enabling "Allow multiple concurrent connections from the same user" I was able to connect the two clients with the same account and your integration perfectly reports the client connections!

YAHOO! BIG Travis :-)

Now I have to create an automation to send me a notification on every connection.
